server/app/src/main/java/com/google/heartrate/wearos/app/gatt/attributes/GattCharacteristic.java

```diff
@@ -0,0 +1,240 @@
+package com.google.heartrate.wearos.app.gatt.attributes;
+
+import android.bluetooth.BluetoothDevice;
+import android.bluetooth.BluetoothGatt;
+import android.bluetooth.BluetoothGattCharacteristic;
+
+import com.google.heartrate.wearos.app.gatt.GattException;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.UUID;
+
+/**
+ * {@link GattCharacteristic} is the wrapper over {@link BluetoothGattCharacteristic} class.
+ * <p>
+ * {@link GattCharacteristic} provides method which are common for all GATT characteristics.
+ * This abstraction makes all operations with {@link BluetoothGattCharacteristic}
+ * expandable for each specific characteristic type.
+ */
+public class GattCharacteristic implements GattAttribute {
+
+    /** {@link BluetoothGattCharacteristic} for characteristic. */
+    private final BluetoothGattCharacteristic mBluetoothGattCharacteristic;
+
+    /** Descriptors which characteristic include. */
+    private final HashMap<UUID, GattDescriptor> descriptorByUuid = new HashMap<>();
```

Comment:
nit: some of your private fields start with m, some don't. I don't mind either way, but consistency would be nice. (We usually don't use the m prefix for private fields)

Reply:
Fixed in [commit](https://github.com/googleinterns/heartrate-bt-wear/pull/1/commits/30619cac7993c70f7f4aa152e5ede9c666fc710c)